src/components/victory-chart/helper-methods.js

```diff
@@ -19,7 +19,7 @@ export default {
     };
 
     if (axisComponents.dependent.length === 0 && axisComponents.independent.length === 0) {
-      return [defaultAxes.independent, defaultAxes.dependent].concat(childComponents);
+      return childComponents.concat([defaultAxes.independent, defaultAxes.dependent]);
```

Comment:
I think it was a preference to have data appear above gridlines by default. I see your point about tick labels though. That does seem more important than whatever is going on with the gridlines.